vtr_flow/arch/xilinx/7series_BRAM_DSP_cary.xml

```diff
@@ -0,0 +1,3396 @@
+      for how these wires connect to each other are not accurate and are loosely based off of the 
+      wilton switch blocks.
+      - This architecture demonstrates the use of L shaped wire types by connecting two separate wires
+      together using a short in the custom switchblocks. It is worth noting that the given channel width 
+      must be fixed at 190 or a multiple of 190 in order to avoid some of the diagonal wires 
+      becoming unconnected.
+      - Stubs (a wires ability to connect to a SB other than the one it terminates in) are modeled in
+      this architecture. For Stubs who connect to SBs parallel to the same direction of travel we use the
+      built in SB pattern feature. For Stubs who connect to SBs perpendicular to the direction of travel
+      we use a L1 wire shorted to the end of the appropriate segment. It is worth noting that we model 
+      all stub types except for stubs of L1 wires.
+      - In Xilinx, and as mentioned in the VTR 9 paper, the concept of how many SB and CB a wire has
+      access to is essentially equal to an equivalent VTR wire of +1 length (i.e. an L1 wire in Xilinx 
+      is equivalent to an L2 in VTR). This is due to VTR having SBs offset from logic elements, while 
+      Xilinx has SBs directly connected to logic elements.
+      - The fc values for CLB inputs and outputs are close approximations to the Xilinx 7-series. 
+      Due to the large number of different wire types in the 7-series we found the VTR struggled to 
+      route designs with low minW when using a fractional fc value (i.e. a fc of 15% connecting to a 
+      wire type that only takes up 6% of the channels has no connections at W < 100). To solve this 
+      we used an absolute fc value which although less accurate, ensures that all wire types have
+      at least some connections at low channel widths. The Fc values given for other logic elements
+      are not accurate and are left for future work.
+
+    Tiles in this architecture: Note that none of the area values for this architecture are accurate.
+    Values for area are pulled from the k6_N10_40nm arch.
+      
+    CLB:
+      - Accurate timing for all elements except for local interconnect (i.e. timing for muxing used to 
+      connect LUT outputs to LUT inputs in the same slice).
+      - Accurate internal structure (LUT, FF, etc.). One exception is that the actual 7-series has a tap
+      point for each COUT of each adder along the carry chain. These where not included due to VTR 
+      causing errors any time they where implemented. Also our CLB capture does not include MUX7 and MUX8
+      modes of the LUTs.
+      - Accurate partial crossbar and feedback connections
+      - Includes special bounce pips which allows CLB inputs and outputs to connect to longer wire types 
+      than the normal L1 and L2.
+    BRAM: 
+      - Acurate timing
+      - Most commonly used modes are implemented with the exception of alternate port width modes on
```

Comment:
alternate port width modes on single port rams -> different widths on the two ports of a simple dual port RAM.

(I don't think you mean single port here)